src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php

```diff
@@ -53,19 +53,23 @@ public function execute(Varien_Event_Observer $observer)
             Mage::register('prevent_observer', true);
         }
     }
-
+    
     /**
      * @return int|false
      */
     protected function getShopgateStoreId()
     {
         /** @var Mage_Core_Model_Store $store */
-        $store = Mage::app()->getStore(self::SHOPGATE_STORE_CODE);
```

Comment:
lol, could have sworn I commented somewhere else on this kind of code just a few weeks ago.
Best to use this code as it does not throw an exception, as far as I remember:
```
$store = Mage::getModel('core/store')->load($storeCode);
```

Reply:
Yepp your're right, Andre and i found a PR from Alex where you commented not to use a Try-Catch here, we weren't quite sure what's the best way to handle this here. Tested it on Casamoda, seems to be working like expected.